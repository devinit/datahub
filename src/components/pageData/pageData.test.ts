import * as prettyFormat from 'pretty-format';
import {
  getCountryProfileData,
  getDistrictProfileData,
  getDistrictProfilePageData,
  getProfilePageData,
  getSpotlightPageData,
  getUnbundlingAidPageData
} from '.';

describe('page data tests', () => {
  it.skip('should return country profile page data', () => {
    const data = getCountryProfileData('uganda');
    expect(prettyFormat(data)).toMatchSnapshot();
  });
  it.skip('should return district page data', () => {
    const data = getDistrictProfileData('wakiso', 'uganda');
    expect(prettyFormat(data)).toMatchSnapshot();
  });
  it.skip('should return spotlightPage page data', () => {
    const data = getSpotlightPageData('uganda');
    expect(prettyFormat(data)).toMatchSnapshot();
  });
  it.skip('should return unbundlingAid page data', () => {
    const data = getUnbundlingAidPageData('oofs');
    expect(prettyFormat(data)).toMatchSnapshot();
  });
  it.skip('should return ProfilePageData page data', () => {
    const data = getProfilePageData('uganda');
    expect(prettyFormat(data)).toMatchSnapshot();
  });
  it('should return DistrictProfilePageData page data', () => {
    const dataA = getDistrictProfilePageData('kenya', 'Nairobi');
    const dataB = getDistrictProfilePageData('uganda', 'Wakiso');
    expect(prettyFormat({ dataA, dataB })).toMatchSnapshot();
  });
});
