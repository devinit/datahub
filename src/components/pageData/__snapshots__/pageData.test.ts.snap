// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page data tests should return DistrictProfilePageData page data 1`] = `
"Object {
  \\"dataA\\": \\"Explore this in-depth profile to find out about poverty, population, education, health, water, sanitation and hygiene, and county public resources in Nairobi.\\",
  \\"dataB\\": \\"Explore this in-depth profile to find out about poverty, population, education, health, water, sanitation and hygiene, and district public resources in Wakiso.\\",
}"
`;

exports[`page data tests should return ProfilePageData page data 1`] = `
"Array [
  Object {
    \\"id\\": \\"spotlight\\",
    \\"narrative\\": \\"Explore this in-depth profile to find out about poverty, population, education, health, water, sanitation and hygiene, and district public resources in {region}.\\",
    \\"title\\": \\"\\",
  },
  Object {
    \\"id\\": \\"not-donor\\",
    \\"narrative\\": \\"Explore this in-depth profile of Uganda to find out overall levels of poverty, income distribution, division of wealth and more. Discover how national and sub-national revenue is generated.\\",
    \\"title\\": \\"\\",
  },
  Object {
    \\"id\\": \\"donor\\",
    \\"narrative\\": \\"Explore this in-depth profile of Uganda to see the international resources it directs to developing countries. Get an overview of government spending, population and income distribution.\\",
    \\"title\\": \\"\\",
  },
]"
`;

exports[`page data tests should return country profile page data 1`] = `
"Array [
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"introduction\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Explore this in-depth profile of Uganda to find out overall levels of poverty, income distribution, division of wealth and more. Discover how national and sub-national revenue is generated\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-ctry-poorest-people\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MANY OF THE POOREST PEOPLE GLOBALLY LIVE IN Uganda?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-ctry-resources\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT RESOURCES ARE AVAILABLE?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"overview-govt-spentperperson\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH DOES THE GOVERNMENT SPEND PER PERSON?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"poverty-reduction-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"IS POVERTY REDUCING OVER TIME?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"poverty-depth-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW DEEP IS POVERTY?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"income-distribution-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS INCOME DISTRIBUTED?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE POPULATION?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-distribution\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE URBAN VS RURAL SPLIT?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"popn-distribution-age-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AGE PROFILE?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"totalrevenue-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE TOTAL REVENUE OF Uganda?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"govt-revenue-external\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH GOVERNMENT REVENUE COMES FROM EXTERNAL GRANTS (AID)?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"ctry-expenditure\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS SPENDING ALLOCATED?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"mixture-of-resources\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT’S THE MIX OF RESOURCES?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"govt-finance-lower\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"How does Uganda finance and spend its budget?\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-A1\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"A dynamic mix of resources flow to and from Uganda. All resources can play a role in ending poverty, though different resources impact poverty directly and indirectly through different mechanisms and flow for wide-ranging objectives.\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-A2\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"To understand the value of different resources for poverty reduction we must start by understanding the volumes - move the slider to explore how official, commercial, private and illicit resources flow to and from Uganda\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"intl-resources-lower-B1\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Where are international resources originating and where are they destined? Start by selecting inflows or outflows:\\",
  },
  Object {
    \\"donor_title\\": \\"\\",
    \\"id\\": \\"avg-income-per-person\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AVERAGE INCOME PER PERSON PER YEAR?\\",
  },
  Object {
    \\"donor_title\\": \\"AS A SHARE OF GNI, HOW MUCH AID DOES Uganda ALLOCATE?\\",
    \\"id\\": \\"share-gni-allocated-to-ctry\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"AS A SHARE OF GNI, HOW MUCH AID IS ALLOCATED TO Uganda?\\",
  },
  Object {
    \\"donor_title\\": \\"HOW HAVE RESOURCE OUTFLOWS CHANGED OVER TIME?\\",
    \\"id\\": \\"resource-inflow\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW HAVE RESOURCE INFLOWS CHANGED OVER TIME?\\",
  },
]"
`;

exports[`page data tests should return district page data 1`] = `
"Array [
  Object {
    \\"id\\": \\"overview-district-poverty\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT PERCENTAGE OF PEOPLE IN Wakiso LIVE BELOW THE NATIONAL POVERTY LINE?\\",
  },
  Object {
    \\"id\\": \\"overview-resources\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT RESOURCES ARE AVAILABLE TO LOCAL GOVERNMENTS IN Wakiso?\\",
  },
  Object {
    \\"id\\": \\"govt-spend-per-person\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH DOES THE LOCAL GOVERNMENT SPEND PER PERSON?\\",
  },
  Object {
    \\"id\\": \\"poverty-levels\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT PERCENTAGE OF PEOPLE IN Wakiso LIVE BELOW THE NATIONAL POVERTY LINE?\\",
  },
  Object {
    \\"id\\": \\"life-expectancy\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AVERAGE LIFE EXPECTANCY?\\",
  },
  Object {
    \\"id\\": \\"std-of-living\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE STANDARD OF LIVING SCORE?\\",
  },
  Object {
    \\"id\\": \\"popn-district\\",
    \\"narrative\\": \\"the total population is ...and the population density is ...per sq km\\",
    \\"title\\": \\"WHAT IS THE POPULATION?\\",
  },
  Object {
    \\"id\\": \\"popn-distribution\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE URBAN VS RURAL SPLIT?\\",
  },
  Object {
    \\"id\\": \\"dependency-ratio\\",
    \\"narrative\\": \\"Compared with the all-district average\\",
    \\"title\\": \\"WHAT IS THE AVERAGE DEPENDENCY RATIO?\\",
  },
  Object {
    \\"id\\": \\"teacher-ratio\\",
    \\"narrative\\": \\"[in government schools and, in other schools]\\",
    \\"title\\": \\"WHAT IS THE PUPIL–TEACHER RATIO IN PRIMARY EDUCATION?\\",
  },
  Object {
    \\"id\\": \\"std-pass-ratio\\",
    \\"narrative\\": \\"and is ranked in \${position} place overall\\",
    \\"title\\": \\"WHAT PERCENTAGE OF STUDENTS PASS THE PRIMARY LEAVING EXAM?\\",
  },
  Object {
    \\"id\\": \\"education-funding\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH PRIMARY EDUCATION FUNDING IS THERE PER PUPIL?\\",
  },
  Object {
    \\"id\\": \\"health-performance\\",
    \\"narrative\\": \\"out of 100, and is ranked in {position} place overall\\",
    \\"title\\": \\"WHAT IS THE DISTRICT LEAGUE HEALTH PERFORMANCE SCORE?\\",
  },
  Object {
    \\"id\\": \\"case-of-tb\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT PERCENTAGE OF TUBERCULOSIS CASES HAVE BEEN SUCCESSFULLY TREATED?\\",
  },
  Object {
    \\"id\\": \\"health-care-funding\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH LOCAL GOVERNMENT HEALTHCARE FUNDING IS THERE PER PERSON?\\",
  },
]"
`;

exports[`page data tests should return spotlightPage page data 1`] = `
"Array [
  Object {
    \\"id\\": \\"comprehensive-source\\",
    \\"narrative\\": \\"is a comprehensive source of Uganda's financial resource flow data at the sub-national (district) level, alongside indicators on poverty, population, education,health, water, hygiene and sanitation. It highlights the geographical variance in sector performance and financial resources, and seeks to answer whether resources are allocated according to need. Explore the country picture by selecting topics and click on a {region} for an in-depth profile.\\",
    \\"title\\": \\"\\",
  },
]"
`;

exports[`page data tests should return unbundlingAid page data 1`] = `"\\" Click the boxes to drill down into OOFs bundle. Drag and drop the buttons below to change the order and take different journeys through the data. Use the ’Compare +’ button to see how different countries’ OOFs bundles compare.\\""`;
